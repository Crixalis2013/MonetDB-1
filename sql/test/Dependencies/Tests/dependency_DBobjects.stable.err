stderr of test 'dependency_DBobjects` in directory 'sql/test/Dependencies` itself:


# 13:01:18 >  
# 13:01:18 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-84217" "--port=36919"
# 13:01:18 >  

MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = CREATE INDEX id_age_index ON v1(id,age);
ERROR = !CREATE INDEX: cannot create index on view 'v1'
CODE  = 42S02
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = ALTER TABLE t1 DROP COLUMN id;
ERROR = !ALTER TABLE: cannot drop column 'id': there are database objects which depend on it
CODE  = 2BM37
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = ALTER TABLE v1 DROP COLUMN age;
ERROR = !ALTER TABLE: cannot drop column from VIEW 'v1'
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = ALTER TABLE t2 ADD FOREIGN KEY(age_v1) REFERENCES v1(age);
ERROR = !CONSTRAINT FOREIGN KEY: could not find referenced PRIMARY KEY in table 'v1'
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = DROP TABLE t1;
ERROR = !DROP TABLE: FOREIGN KEY t2.t2_id_t1_fkey depends on t1
CODE  = 40000
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = DROP VIEW v1;
ERROR = !DROP VIEW: cannot drop view 'v1', there are database objects which depend on it
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = DROP TABLE t2; 
ERROR = !DROP TABLE: unable to drop table t2 (there are database objects which depend on it)
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = DROP FUNCTION f1;
ERROR = !DROP FUNCTION: there are database objects dependent on function f1;
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = DROP INDEX id_age_index;
ERROR = !DROP INDEX: no such index 'id_age_index'
CODE  = 42S12
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = DROP VIEW v1;
ERROR = !DROP VIEW: cannot drop view 'v1', there are database objects which depend on it
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-84217/.s.monetdb.36919
QUERY = DROP TABLE t1;
ERROR = !DROP TABLE: unable to drop table t1 (there are database objects which depend on it)
CODE  = 42000

# 13:01:18 >  
# 13:01:18 >  "Done."
# 13:01:18 >  

