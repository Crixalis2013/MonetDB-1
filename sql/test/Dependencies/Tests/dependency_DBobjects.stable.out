stdout of test 'dependency_DBobjects` in directory 'sql/test/Dependencies` itself:


# 13:01:18 >  
# 13:01:18 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-84217" "--port=36919"
# 13:01:18 >  

#CREATE TABLE t1(id int, name varchar(1024), age int, PRIMARY KEY(id));
#CREATE VIEW v1 as select id, age from t1 where name like 'monet%';
#CREATE VIEW v2 as select * from v1;
#CREATE FUNCTION f1(num int)
#RETURNS int
#BEGIN 
#	IF num >0
#		THEN RETURN 1;
#		ELSE RETURN 0;
#	END IF;
#END;
#CREATE FUNCTION f2()
#RETURNS TABLE(id int)
#BEGIN
#	DECLARE TABLE f1_t1(id int);
#	DECLARE x int;
#	
#	SET x = 3;	
#
#	IF f1(x) > 0
#		THEN RETURN f1_t1;
#	END IF;
#
#	INSERT INTO f1_t1 VALUES(1);
#
#	IF f1(x) < 0
#		THEN RETURN f1_t1;
#	END IF;
#
#	RETURN TABLE (SELECT t1.id FROM v1, t1 WHERE v1.age > 10 AND t1.name LIKE 'monet');
#END;
#CREATE INDEX id_index ON t1(id);
#CREATE TABLE t2 (id_t1 int, age_v1 int);
#ALTER TABLE t2 ADD FOREIGN KEY(id_t1) REFERENCES t1(id);
#CREATE TRIGGER trigger_test AFTER INSERT ON t1
#	INSERT INTO t2 values(1,23);
#CREATE TABLE t3 (id int);
#CREATE TRIGGER trigger_test_4 AFTER INSERT ON t1
#	INSERT INTO t3 values(1);
#SELECT s.name, u.name, 'DEP_USER' from sys.schemas as s, users u where u.default_schema = s.id order by s.name, u.name;
% sys.s,	.u,	.%1 # table_name
% name,	name,	%1 # name
% varchar,	varchar,	char # type
% 4,	10,	8 # length
[ "sys",	".snapshot",	"DEP_USER"	]
[ "sys",	"monetdb",	"DEP_USER"	]
[ "test",	"monet_test",	"DEP_USER"	]
#SELECT a.name, s.name, 'DEP_SCHEMA' from sys.schemas as s, auths a where s.owner = a.id order by a.name, s.name;
% sys.a,	sys.s,	.%1 # table_name
% name,	name,	%1 # name
% varchar,	varchar,	char # type
% 7,	8,	10 # length
[ "monetdb",	"json",	"DEP_SCHEMA"	]
[ "monetdb",	"logging",	"DEP_SCHEMA"	]
[ "monetdb",	"profiler",	"DEP_SCHEMA"	]
[ "monetdb",	"sys",	"DEP_SCHEMA"	]
[ "monetdb",	"test",	"DEP_SCHEMA"	]
[ "monetdb",	"tmp",	"DEP_SCHEMA"	]
[ "monetdb",	"wlc",	"DEP_SCHEMA"	]
[ "monetdb",	"wlr",	"DEP_SCHEMA"	]
#SELECT t.name, v.name, 'DEP_VIEW' from sys.tables as t, sys.tables as v, sys.dependencies as dep where t.id = dep.id AND v.id = dep.depend_id AND dep.depend_type = 5 AND v.type = 1 order by t.name, v.name;
% .t,	.v,	.%17 # table_name
% name,	name,	%17 # name
% varchar,	varchar,	char # type
% 2,	2,	8 # length
[ "t1",	"v1",	"DEP_VIEW"	]
[ "v1",	"v2",	"DEP_VIEW"	]
#SELECT t.name, i.name, 'DEP_INDEX' from sys.tables as t, sys.idxs as i where i.table_id = t.id and i.name not in (select name from sys.keys) and t.type = 0 order by t.name, i.name;
% .t,	.i,	.%13 # table_name
% name,	name,	%13 # name
% varchar,	varchar,	char # type
% 2,	8,	9 # length
[ "t1",	"id_index",	"DEP_INDEX"	]
#(SELECT t.name as name, tri.name as trigname, 'DEP_TRIGGER' from sys.tables as t, sys.triggers as tri where tri.table_id = t.id) UNION (SELECT t.name as name, tri.name as trigname, 'DEP_TRIGGER' from sys.triggers tri, sys.tables t, sys.dependencies dep where dep.id = t.id AND dep.depend_id =tri.id AND dep.depend_type = 8) order by name, trigname;
% .%25,	.%25,	.%25 # table_name
% name,	trigname,	%20 # name
% varchar,	varchar,	char # type
% 7,	21,	11 # length
[ "_tables",	"system_update_tables",	"DEP_TRIGGER"	]
[ "schemas",	"system_update_schemas",	"DEP_TRIGGER"	]
[ "t1",	"trigger_test",	"DEP_TRIGGER"	]
[ "t1",	"trigger_test_4",	"DEP_TRIGGER"	]
[ "t2",	"trigger_test",	"DEP_TRIGGER"	]
[ "t3",	"trigger_test_4",	"DEP_TRIGGER"	]
#SELECT t.name, fk.name, 'DEP_FKEY' from sys.tables as t, sys.keys as k, sys.keys as fk where fk.rkey = k.id and k.table_id = t.id order by t.name, fk.name;
% .t,	.fk,	.%10 # table_name
% name,	name,	%10 # name
% varchar,	varchar,	char # type
% 2,	13,	8 # length
[ "t1",	"t2_id_t1_fkey",	"DEP_FKEY"	]
#SELECT t.name, f.name, 'DEP_FUNC' from sys.functions as f, sys.tables as t, sys.dependencies as dep where t.id = dep.id AND f.id = dep.depend_id AND dep.depend_type = 7 AND t.type IN (0, 10) ORDER BY t.name, f.name;
% .t,	.f,	.%13 # table_name
% name,	name,	%13 # name
% varchar,	varchar,	char # type
% 18,	17,	8 # length
[ "_columns",	"describe_columns",	"DEP_FUNC"	]
[ "_tables",	"describe_columns",	"DEP_FUNC"	]
[ "_tables",	"describe_table",	"DEP_FUNC"	]
[ "comments",	"describe_columns",	"DEP_FUNC"	]
[ "comments",	"describe_function",	"DEP_FUNC"	]
[ "comments",	"describe_table",	"DEP_FUNC"	]
[ "function_languages",	"describe_function",	"DEP_FUNC"	]
[ "function_types",	"describe_function",	"DEP_FUNC"	]
[ "functions",	"describe_function",	"DEP_FUNC"	]
[ "schemas",	"describe_columns",	"DEP_FUNC"	]
[ "schemas",	"describe_function",	"DEP_FUNC"	]
[ "schemas",	"describe_table",	"DEP_FUNC"	]
[ "spatial_ref_sys",	"getproj4",	"DEP_FUNC"	]
[ "t1",	"f2",	"DEP_FUNC"	]
[ "table_types",	"describe_table",	"DEP_FUNC"	]
#SELECT c.name, v.name, 'DEP_VIEW' from sys.columns as c, sys.tables as v, sys.dependencies as dep where c.id = dep.id AND v.id = dep.depend_id AND dep.depend_type = 5 AND v.type = 1 order by c.name, v.name;
% .c,	.v,	.%15 # table_name
% name,	name,	%15 # name
% varchar,	varchar,	char # type
% 4,	2,	8 # length
[ "age",	"v1",	"DEP_VIEW"	]
[ "age",	"v2",	"DEP_VIEW"	]
[ "id",	"v1",	"DEP_VIEW"	]
[ "id",	"v2",	"DEP_VIEW"	]
[ "name",	"v1",	"DEP_VIEW"	]
#SELECT c.name, k.name, 'DEP_KEY' from sys.columns as c, sys.objects as kc, sys.keys as k where kc."name" = c.name AND kc.id = k.id AND k.table_id = c.table_id AND k.rkey = -1 order by c.name, k.name;
% .c,	.k,	.%6 # table_name
% name,	name,	%6 # name
% varchar,	varchar,	char # type
% 20,	44,	7 # length
[ "dependency_type_id",	"dependency_types_dependency_type_id_pkey",	"DEP_KEY"	]
[ "dependency_type_name",	"dependency_types_dependency_type_name_unique",	"DEP_KEY"	]
[ "function_type_id",	"function_types_function_type_id_pkey",	"DEP_KEY"	]
[ "function_type_name",	"function_types_function_type_name_unique",	"DEP_KEY"	]
[ "id",	"comments_id_pkey",	"DEP_KEY"	]
[ "id",	"t1_id_pkey",	"DEP_KEY"	]
[ "index_type_id",	"index_types_index_type_id_pkey",	"DEP_KEY"	]
[ "index_type_name",	"index_types_index_type_name_unique",	"DEP_KEY"	]
[ "key_type_id",	"key_types_key_type_id_pkey",	"DEP_KEY"	]
[ "key_type_name",	"key_types_key_type_name_unique",	"DEP_KEY"	]
[ "keyword",	"keywords_keyword_pkey",	"DEP_KEY"	]
[ "language_id",	"function_languages_language_id_pkey",	"DEP_KEY"	]
[ "language_name",	"function_languages_language_name_unique",	"DEP_KEY"	]
[ "privilege_code_id",	"privilege_codes_privilege_code_id_pkey",	"DEP_KEY"	]
[ "privilege_code_name",	"privilege_codes_privilege_code_name_unique",	"DEP_KEY"	]
[ "srid",	"spatial_ref_sys_srid_pkey",	"DEP_KEY"	]
[ "table_type_id",	"table_types_table_type_id_pkey",	"DEP_KEY"	]
[ "table_type_name",	"table_types_table_type_name_unique",	"DEP_KEY"	]
#SELECT c.name, i.name, 'DEP_INDEX' from sys.columns as c, sys.objects as kc, sys.idxs as i where kc."name" = c.name AND kc.id = i.id AND c.table_id = i.table_id AND i.name not in (select name from sys.keys) order by c.name, i.name;
% .c,	.i,	.%11 # table_name
% name,	name,	%11 # name
% varchar,	varchar,	char # type
% 2,	8,	9 # length
[ "id",	"id_index",	"DEP_INDEX"	]
#SELECT c.name, f.name, 'DEP_FUNC' from sys.functions as f, sys.columns as c, sys.dependencies as dep where c.id = dep.id AND f.id = dep.depend_id AND dep.depend_type = 7 ORDER BY c.name, f.name;
% .c,	.f,	.%6 # table_name
% name,	name,	%6 # name
% varchar,	varchar,	char # type
% 21,	17,	8 # length
[ "access",	"describe_columns",	"DEP_FUNC"	]
[ "access",	"describe_table",	"DEP_FUNC"	]
[ "age",	"f2",	"DEP_FUNC"	]
[ "age",	"f2",	"DEP_FUNC"	]
[ "auth_name",	"getproj4",	"DEP_FUNC"	]
[ "auth_srid",	"getproj4",	"DEP_FUNC"	]
[ "authorization",	"describe_columns",	"DEP_FUNC"	]
[ "authorization",	"describe_function",	"DEP_FUNC"	]
[ "authorization",	"describe_table",	"DEP_FUNC"	]
[ "commit_action",	"describe_columns",	"DEP_FUNC"	]
[ "commit_action",	"describe_table",	"DEP_FUNC"	]
[ "default",	"describe_columns",	"DEP_FUNC"	]
[ "func",	"describe_function",	"DEP_FUNC"	]
[ "function_type_id",	"describe_function",	"DEP_FUNC"	]
[ "function_type_keyword",	"describe_function",	"DEP_FUNC"	]
[ "function_type_name",	"describe_function",	"DEP_FUNC"	]
[ "id",	"describe_columns",	"DEP_FUNC"	]
[ "id",	"describe_columns",	"DEP_FUNC"	]
[ "id",	"describe_columns",	"DEP_FUNC"	]
[ "id",	"describe_columns",	"DEP_FUNC"	]
[ "id",	"describe_function",	"DEP_FUNC"	]
[ "id",	"describe_function",	"DEP_FUNC"	]
[ "id",	"describe_function",	"DEP_FUNC"	]
[ "id",	"describe_table",	"DEP_FUNC"	]
[ "id",	"describe_table",	"DEP_FUNC"	]
[ "id",	"describe_table",	"DEP_FUNC"	]
[ "id",	"f2",	"DEP_FUNC"	]
[ "id",	"f2",	"DEP_FUNC"	]
[ "language",	"describe_function",	"DEP_FUNC"	]
[ "language_id",	"describe_function",	"DEP_FUNC"	]
[ "language_keyword",	"describe_function",	"DEP_FUNC"	]
[ "language_name",	"describe_function",	"DEP_FUNC"	]
[ "mod",	"describe_function",	"DEP_FUNC"	]
[ "name",	"describe_columns",	"DEP_FUNC"	]
[ "name",	"describe_columns",	"DEP_FUNC"	]
[ "name",	"describe_columns",	"DEP_FUNC"	]
[ "name",	"describe_function",	"DEP_FUNC"	]
[ "name",	"describe_function",	"DEP_FUNC"	]
[ "name",	"describe_table",	"DEP_FUNC"	]
[ "name",	"describe_table",	"DEP_FUNC"	]
[ "name",	"f2",	"DEP_FUNC"	]
[ "null",	"describe_columns",	"DEP_FUNC"	]
[ "number",	"describe_columns",	"DEP_FUNC"	]
[ "owner",	"describe_columns",	"DEP_FUNC"	]
[ "owner",	"describe_function",	"DEP_FUNC"	]
[ "owner",	"describe_table",	"DEP_FUNC"	]
[ "proj4text",	"getproj4",	"DEP_FUNC"	]
[ "query",	"describe_columns",	"DEP_FUNC"	]
[ "query",	"describe_table",	"DEP_FUNC"	]
[ "remark",	"describe_columns",	"DEP_FUNC"	]
[ "remark",	"describe_function",	"DEP_FUNC"	]
[ "remark",	"describe_table",	"DEP_FUNC"	]
[ "schema_id",	"describe_columns",	"DEP_FUNC"	]
[ "schema_id",	"describe_function",	"DEP_FUNC"	]
[ "schema_id",	"describe_table",	"DEP_FUNC"	]
[ "semantics",	"describe_function",	"DEP_FUNC"	]
[ "side_effect",	"describe_function",	"DEP_FUNC"	]
[ "srid",	"getproj4",	"DEP_FUNC"	]
[ "srtext",	"getproj4",	"DEP_FUNC"	]
[ "storage",	"describe_columns",	"DEP_FUNC"	]
[ "system",	"describe_columns",	"DEP_FUNC"	]
[ "system",	"describe_columns",	"DEP_FUNC"	]
[ "system",	"describe_function",	"DEP_FUNC"	]
[ "system",	"describe_function",	"DEP_FUNC"	]
[ "system",	"describe_table",	"DEP_FUNC"	]
[ "system",	"describe_table",	"DEP_FUNC"	]
[ "table_id",	"describe_columns",	"DEP_FUNC"	]
[ "table_type_id",	"describe_table",	"DEP_FUNC"	]
[ "table_type_name",	"describe_table",	"DEP_FUNC"	]
[ "type",	"describe_columns",	"DEP_FUNC"	]
[ "type",	"describe_columns",	"DEP_FUNC"	]
[ "type",	"describe_function",	"DEP_FUNC"	]
[ "type",	"describe_table",	"DEP_FUNC"	]
[ "type_digits",	"describe_columns",	"DEP_FUNC"	]
[ "type_scale",	"describe_columns",	"DEP_FUNC"	]
[ "vararg",	"describe_function",	"DEP_FUNC"	]
[ "varres",	"describe_function",	"DEP_FUNC"	]
#SELECT c.name, tri.name, 'DEP_TRIGGER' from sys.columns as c, sys.triggers as tri, sys.dependencies as dep where dep.id = c.id AND dep.depend_id =tri.id AND dep.depend_type = 8 order by c.name, tri.name;
% .c,	.tri,	.%6 # table_name
% name,	name,	%6 # name
% varchar,	varchar,	char # type
% 6,	14,	11 # length
[ "age",	"trigger_test",	"DEP_TRIGGER"	]
[ "age_v1",	"trigger_test",	"DEP_TRIGGER"	]
[ "id",	"trigger_test",	"DEP_TRIGGER"	]
[ "id",	"trigger_test_4",	"DEP_TRIGGER"	]
[ "id_t1",	"trigger_test",	"DEP_TRIGGER"	]
[ "name",	"trigger_test",	"DEP_TRIGGER"	]
#SELECT v.name, f.name, 'DEP_FUNC' from sys.functions as f, sys.tables as v, sys.dependencies as dep where v.id = dep.id AND f.id = dep.depend_id AND dep.depend_type = 7 AND v.type IN (1, 11) ORDER BY v.name, f.name;
% .v,	.f,	.%13 # table_name
% name,	name,	%13 # name
% varchar,	varchar,	char # type
% 2,	2,	8 # length
[ "v1",	"f2",	"DEP_FUNC"	]
#SELECT v.name, i.name, 'DEP_INDEX' from sys.tables as v, sys.idxs as i where i.table_id = v.id and i.name not in (select name from sys.keys) and v.type = 1 order by v.name, i.name;
% .v,	.i,	.%13 # table_name
% name,	name,	%13 # name
% varchar,	varchar,	char # type
% 0,	0,	9 # length
#SELECT v.name, tri.name, 'DEP_TRIGGER' from sys.tables as v, sys.triggers as tri, sys.dependencies as dep where dep.id = v.id AND dep.depend_id =tri.id AND dep.depend_type = 8 AND v.type = 1 order by v.name, tri.name;
% .v,	.tri,	.%10 # table_name
% name,	name,	%10 # name
% varchar,	varchar,	char # type
% 0,	0,	11 # length
#SELECT f1.name, f2.name, 'DEP_FUNC' from sys.functions as f1, sys.functions as f2, sys.dependencies as dep where f1.id = dep.id AND f2.id = dep.depend_id AND dep.depend_type = 7 order by f2.name, f1.name;
% sys.f1,	sys.f2,	.%1 # table_name
% name,	name,	%1 # name
% varchar,	varchar,	char # type
% 17,	16,	8 # length
[ "describe_type",	"describe_columns",	"DEP_FUNC"	]
[ "f1",	"f2",	"DEP_FUNC"	]
[ "ms_trunc",	"ms_round",	"DEP_FUNC"	]
[ "st_numgeometries",	"st_numpatches",	"DEP_FUNC"	]
[ "st_geometryn",	"st_patchn",	"DEP_FUNC"	]
[ "getproj4",	"st_transform",	"DEP_FUNC"	]
[ "internaltransform",	"st_transform",	"DEP_FUNC"	]
[ "st_srid",	"st_transform",	"DEP_FUNC"	]
#SELECT f.name, tri.name, 'DEP_TRIGGER' from sys.functions as f, sys.triggers as tri, sys.dependencies as dep where dep.id = f.id AND dep.depend_id =tri.id AND dep.depend_type = 8 order by f.name, tri.name;
% sys.f,	sys.tri,	.%1 # table_name
% name,	name,	%1 # name
% varchar,	varchar,	char # type
% 0,	0,	11 # length
#SELECT k.name, fk.name, 'DEP_FKEY' from sys.keys as k, sys.keys as fk where fk.rkey = k.id order by k.name, fk.name;
% sys.k,	sys.fk,	.%1 # table_name
% name,	name,	%1 # name
% varchar,	varchar,	char # type
% 10,	13,	8 # length
[ "t1_id_pkey",	"t2_id_t1_fkey",	"DEP_FKEY"	]
#DROP FUNCTION f2;
#DROP TRIGGER trigger_test;
#DROP INDEX id_index;
#DROP TABLE t2; 
#DROP FUNCTION f1;
#DROP VIEW v2;
#DROP VIEW v1;
#DROP TRIGGER trigger_test_4;
#DROP TABLE t1;
#DROP TABLE t3;

# 13:01:18 >  
# 13:01:18 >  "Done."
# 13:01:18 >  

